### ISSUE TITLE
Navigation editor UI tweaks

### ISSUE THREAD

**Issue by mikemurray** [reactioncommerce/reaction#5163]
## Issue Description

The **Edit Navigation Item** modal may hide the action buttons on some smaller screen sizes.

![image](https://user-images.githubusercontent.com/42217/57094554-aa1e9980-6cc5-11e9-81d8-f5c324c654b4.png)

**Saving/publishing** navigation items give no indication of success or failure. It's also impossible to tell if there are any unpublished changes.

![Not-found](https://user-images.githubusercontent.com/42217/57094814-43e64680-6cc6-11e9-98ac-fab4d0c8ef91.png)

## Possible Solution

- Edit dialog should be split into header/content/actions with the content region being the only scrollable one
- Add "Has unpublished changes" next to the discard button at the very least
- Have the `Save changes` buttons indicate that it is `[spinner] Saving changes` when clicked
- Display an alert toast when there are errors
- Display an indication of published status on the navigation tree items. (This may not be easily achieved)

**Comment by spencern** [comment]
@mikemurray can we close this now?

**Comment by mikemurray** [comment]
No. This is additional stuff not tackled in the other navigation PR.
